sdk/eventhub/event-hubs/src/managementClient.ts

```diff
@@ -306,31 +310,145 @@ export class ManagementClient extends LinkEntity {
-        abortSignal: options.abortSignal,
-        requestName: options.requestName,
-        timeoutInSeconds: getRetryAttemptTimeoutInMs(options.retryOptions) / 1000,
-        delayInSeconds:
-          options.retryOptions &&
-          options.retryOptions.retryInterval &&
-          options.retryOptions.retryInterval >= 0
-            ? options.retryOptions.retryInterval / 1000
-            : undefined
+      const aborter: AbortSignalLike | undefined = options && options.abortSignal;
+
+      const sendOperationPromise = () =>
+        new Promise<Message>(async (resolve, reject) => {
+          try {
+            let count = 0;
+
+            if (!options) {
+              options = {};
+            }
+
+            let timeOver: boolean = false;
+            const retryTimeoutInMs = getRetryAttemptTimeoutInMs(options.retryOptions);
+            let timeTakenByInit = 0;
+
+            if (!this._isMgmtRequestResponseLinkOpen()) {
+              log.mgmt(
+                "[%s] Acquiring lock to get the management req res link.",
+                this._context.connectionId
+              );
+
+              const rejectOnAbort = () => {
+                const address = this._mgmtReqResLink!.receiver.address || "address";
+                const requestName = options!.requestName;
+                const desc: string =
+                  `[${this._context.connectionId}] The request "${requestName}" ` +
+                  `to "${address}" has been cancelled by the user.`;
+                log.error(desc);
+                const error = new AbortError(
+                  `The ${
+                    requestName ? requestName + " " : ""
+                  }operation has been cancelled by the user.`
+                );
+
+                return reject(error);
+              };
+
+              const onAbort = () => {
+                // safe to clear the timeout if it hasn't already occurred.
+                if (!timeOver) {
+                  clearTimeout(waitTimer);
+                }
+                aborter!.removeEventListener("abort", onAbort);
+
+                rejectOnAbort();
+              };
+
+              if (aborter) {
+                if (aborter.aborted) {
+                  return rejectOnAbort();
+                }
+                aborter.addEventListener("abort", onAbort);
+              }
+
+              const initOperationStartTime = Date.now();
+
+              const actionAfterTimeout = () => {
+                timeOver = true;
+                const address = this._mgmtReqResLink!.receiver.address || "address";
+                const desc: string =
+                  `The request with message_id "${request.message_id}" to "${address}" ` +
+                  `endpoint timed out. Please try again later.`;
+                const e: Error = {
+                  name: "OperationTimeoutError",
+                  message: desc
+                };
+                return reject(translate(e));
```

Comment:
We need to remove the event listener for abort event here before returning with the rejected promise

Reply:
Updated

Reply:
I recall this being resolved, but now I don't see the removal of event listener when the timer times out.

Reply:
We aren't adding listeners anymore as part of the outer abort for init.
Asynchronous abortion (by using event emitter listeners), is currently not trivial for `init` and right way to go about it would be to pass in the aborter to `init` 
So that the connection gets cleanly closed if already in progress of being created.

Discussed this with @chradek (See https://github.com/Azure/azure-sdk-for-js/pull/4322#discussion_r307429438)
Since abort during init is not solved for receive as well, and needs more work and tests, we were looking to move it out of this PR.